docs/how-to/partitions.rst

```diff
@@ -0,0 +1,136 @@
+******************************************
+Add partition support to an application
+******************************************
+
+Partitions basics
+=================
+
+In the below examples, we work with three partitions: ``default``, ``kernel``, and ``bar-baz``.  The ``default`` and ``kernel`` partitions do not have a namespace.  The ``bar-baz`` partition is part of the ``component`` namespace.
+
+``default`` must always be the first listed partition.
+
+Partition names and namespace names must consist of *only* lower-case alphabetic characters, unless a partition exists under a namespace, in which case it may also contain hyphen characters, though the first and last characters must still be alphabetic.
+
+.. _app_changes:
+
+Required application changes
+============================
+
+To add partition support to an application, two basic changes are needed to the ``craft-application``.
+
+#. Enable the feature
+
+   When creating the :class:`AppMetadata <craft_application.AppMetadata>`, specify that the application will use the partitions feature:
+
+   .. code-block:: python
+
+     from craft_application import AppMetadata, AppFeatures
+
+     APP_METADATA = AppMetadata(
+       ...
+       features=AppFeatures(enable_partitions=True),
+     )
+
+#. Define the list of partitions
+
+   We need to tell the :class:`LifecycleManager <craft_parts.LifecycleManager>` class about our partitions, but applications do not usually directly instantiate the LifecycleManager.
+
+   Inside your :class:`Application <craft_application.Application>` class, the :class:`ServiceFactory <craft_application.ServiceFactory>` can be told to instantiate the :class:`LifecycleService <craft_application.LifecycleService` with specific ``custom_args`` by calling ``set_kwargs``.
+
+   Override the ``_configure_services`` method, and use ``set_kwargs`` to define the partitions that will eventually be passed to the :class:`LifecycleManager <craft_parts.LifecycleManager>`:
+
+   .. code-block:: python
+
+     @override
+     def _configure_services(self, provider_name: str | None) -> None:
+       super()._configure_services(provider_name)
+
+       ...
+
+       self.services.set_kwargs(
+         "lifecycle",
+         ...
+         partitions=["default", "kernel", "component/bar-baz"],
+       )
+
+Using the partitions
+====================
+
+Partitions cannot be used until `after you have configured your application <#app-changes>`_.
+
+In configuration yaml lifecycle
+-------------------------------
```

Comment:
This section title is a bit confusing, maybe it could be something like "In the parts definition"? Note that craft-parts doesn't have a "project" concept.

Reply:
Will update section title in the `craft-application` partitions docs PR.